Runtime/editor/editor_mouse_tracking.cs

```diff
@@ -0,0 +1,189 @@
+using System.Collections.Generic;
+using Unity.UIWidgets.foundation;
+using Unity.UIWidgets.scheduler;
+using UnityEditor;
+
+namespace Unity.UIWidgets.gestures {
+    public partial class MouseTracker {
+        bool _enableDragFromEditorRelease = false;
+
+        void _handleDragFromEditorEvent(PointerEvent evt, int deviceId) {
+            if (evt is PointerDragFromEditorReleaseEvent) {
+                this._enableDragFromEditorRelease = false;
+                this._scheduleDragFromEditorReleaseCheck();
+                this._lastMouseEvent.Remove(deviceId);
+            }
+            else if (evt is PointerDragFromEditorEnterEvent ||
+                     evt is PointerDragFromEditorHoverEvent ||
+                     evt is PointerDragFromEditorExitEvent) {
+                if (!this._lastMouseEvent.ContainsKey(deviceId) ||
+                    this._lastMouseEvent[deviceId].position != evt.position) {
+                    // Only schedule a frame if we have our first event, or if the
+                    // location of the mouse has changed, and only if there are tracked annotations.
+                    this._scheduleDragFromEditorMousePositionCheck();
+                }
+
+                this._lastMouseEvent[deviceId] = evt;
+            }
+        }
+
+        void _scheduleDragFromEditorReleaseCheck() {
+            DragAndDrop.AcceptDrag();
+
+            var lastMouseEvent = new List<(PointerEvent, int)>();
+            foreach (int deviceId in this._lastMouseEvent.Keys) {
+                lastMouseEvent.Add((this._lastMouseEvent[deviceId], deviceId));
+            }
+
+            SchedulerBinding.instance.addPostFrameCallback(_ => {
+                foreach (var lastEvent in lastMouseEvent) {
+                    MouseTrackerAnnotation hit = this.annotationFinder(lastEvent.Item1.position);
+
+                    if (hit == null) {
+                        foreach (_TrackedAnnotation trackedAnnotation in this._trackedAnnotations.Values) {
+                            if (trackedAnnotation.activeDevices.Contains(lastEvent.Item2)) {
+                                trackedAnnotation.activeDevices.Remove(lastEvent.Item2);
+                            }
+                        }
+
+                        return;
+                    }
+
+                    _TrackedAnnotation hitAnnotation = this._findAnnotation(hit);
+
+                    // release
+                    if (hitAnnotation.activeDevices.Contains(lastEvent.Item2)) {
```

Comment:
Item1, Item2 are not appropriate names, please try to rename these properties properly so that any reader can easily understand their meanings